ktor-server/ktor-server-plugins/ktor-server-freemarker/jvm/src/io/ktor/server/freemarker/FreeMarker.kt

```diff
@@ -45,11 +47,24 @@ public val FreeMarker: ApplicationPlugin<Configuration> = createApplicationPlugi
         return result
     }
 
-    onCallRespond { _, message ->
+    on(ResponseBeforeTransform) { message ->
         if (message is FreeMarkerContent) {
             transformBody {
                 process(message)
             }
         }
     }
 }
+
+internal object ResponseBeforeTransform : Hook<suspend PipelineContext<Any, ApplicationCall>.(body: Any) -> Any> {
```

Comment:
A hook can have generic of `suspend (body: Any, type: TypeInfo) -> Any`. Then you won't need `tranformBody` block and can just return your value from `on(MyHook) { body, type -> process(body) }`